Crate is broken due to netlink-* bumps

Looks like a recent netlink dependency bump breaks build of the `defguard_wireguard_rs` crate:
```
error[E0277]: the trait bound `netlink_packet_wireguard::Wireguard: GenlFamily` is not satisfied
...
note: there are multiple different versions of crate `netlink_packet_generic` in the dependency graph
...
```

Also reported upstream as https://github.com/rust-netlink/netlink-packet-wireguard/issues/13.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -23,8 +23,6 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v5
-        with:
-          submodules: recursive
 
       - name: Cache
         uses: Swatinem/rust-cache@v2

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -18,24 +18,28 @@ env:
 jobs:
   test:
     runs-on: [self-hosted, Linux, X64]
-    container: rust:1.80
+    container: rust:1-slim
 
     steps:
-      - name: Debug
-        run: echo ${{ github.ref_name }}
       - name: Checkout
         uses: actions/checkout@v3
         with:
           submodules: recursive
+
       - name: Cache
         uses: Swatinem/rust-cache@v2
         with:
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
       - name: Check format
         run: |
           rustup component add rustfmt
           cargo fmt -- --check
+
       - name: Run cargo deny
-        uses: EmbarkStudios/cargo-deny-action@v2
+        run: |
+          cargo install cargo-deny
+          cargo deny check
+
       - name: Run tests
         run: cargo test --locked --no-fail-fast

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -22,7 +22,7 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v5
         with:
           submodules: recursive
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "defguard_wireguard_rs"
-version = "0.7.5"
+version = "0.7.6"
 edition = "2021"
 rust-version = "1.80"
 description = "A unified multi-platform high-level API for managing WireGuard interfaces"
@@ -27,10 +27,10 @@ libc = { version = "0.2", default-features = false }
 nix = { version = "0.30", features = ["ioctl", "socket"] }
 
 [target.'cfg(target_os = "linux")'.dependencies]
-netlink-packet-core = "0.7"
-netlink-packet-generic = "0.3"
-netlink-packet-route = "0.22"
-netlink-packet-utils = "0.5"
+netlink-packet-core = "0.8"
+netlink-packet-generic = "0.4"
+netlink-packet-route = "0.25"
+netlink-packet-utils = "0.6"
 netlink-packet-wireguard = "0.2"
 netlink-sys = "0.8"
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,8 +1,8 @@
 [package]
 name = "defguard_wireguard_rs"
 version = "0.7.6"
-edition = "2021"
-rust-version = "1.80"
+edition = "2024"
+rust-version = "1.85"
 description = "A unified multi-platform high-level API for managing WireGuard interfaces"
 license = "Apache-2.0"
 readme = "README.md"

--- a/examples/client.rs
+++ b/examples/client.rs
@@ -1,8 +1,8 @@
 use std::{net::SocketAddr, str::FromStr};
 
 use defguard_wireguard_rs::{
-    host::Peer, key::Key, net::IpAddrMask, InterfaceConfiguration, Kernel, WGApi,
-    WireguardInterfaceApi,
+    InterfaceConfiguration, Kernel, WGApi, WireguardInterfaceApi, host::Peer, key::Key,
+    net::IpAddrMask,
 };
 use x25519_dalek::{EphemeralSecret, PublicKey};
 

--- a/examples/server.rs
+++ b/examples/server.rs
@@ -1,8 +1,8 @@
 use std::str::FromStr;
 
 use defguard_wireguard_rs::{
-    host::Peer, key::Key, net::IpAddrMask, InterfaceConfiguration, Kernel, WGApi,
-    WireguardInterfaceApi,
+    InterfaceConfiguration, Kernel, WGApi, WireguardInterfaceApi, host::Peer, key::Key,
+    net::IpAddrMask,
 };
 use x25519_dalek::{EphemeralSecret, PublicKey};
 

--- a/examples/userspace.rs
+++ b/examples/userspace.rs
@@ -1,5 +1,5 @@
 #[cfg(target_os = "macos")]
-use std::io::{stdin, stdout, Read, Write};
+use std::io::{Read, Write, stdin, stdout};
 
 #[cfg(target_os = "macos")]
 use defguard_wireguard_rs::{Userspace, WGApi, WireguardInterfaceApi};

--- a/src/bsd/ifconfig.rs
+++ b/src/bsd/ifconfig.rs
@@ -3,13 +3,12 @@ use std::{
     os::fd::AsRawFd,
 };
 
-use libc::{IFF_UP, IF_NAMESIZE};
+use libc::{IF_NAMESIZE, IFF_UP};
 use nix::{ioctl_readwrite, ioctl_write_ptr, sys::socket::AddressFamily};
 
 use super::{
-    create_socket,
+    IoError, create_socket,
     sockaddr::{SockAddrIn, SockAddrIn6},
-    IoError,
 };
 
 // From `netinet6/in6.h`.

--- a/src/bsd/mod.rs
+++ b/src/bsd/mod.rs
@@ -8,7 +8,7 @@ mod wgio;
 use std::{
     collections::HashMap,
     ffi::{CStr, CString},
-    mem::{size_of, MaybeUninit},
+    mem::{MaybeUninit, size_of},
     net::IpAddr,
     os::fd::OwnedFd,
     ptr::from_ref,
@@ -17,7 +17,7 @@ use std::{
 
 use nix::{
     errno::Errno,
-    sys::socket::{socket, AddressFamily, SockFlag, SockType},
+    sys::socket::{AddressFamily, SockFlag, SockType, socket},
 };
 use route::{DestAddrMask, GatewayLink};
 use sockaddr::{SockAddrDl, SockAddrIn, SockAddrIn6, SocketFromRaw};
@@ -32,9 +32,9 @@ use self::{
     wgio::{WgReadIo, WgWriteIo},
 };
 use crate::{
+    IpVersion, Key, WireguardInterfaceError,
     host::{Host, Peer},
     net::IpAddrMask,
-    IpVersion, Key, WireguardInterfaceError,
 };
 
 // Note: these values differ across different platforms.
@@ -67,14 +67,14 @@ static NV_IPV4: &str = "ipv4";
 static NV_IPV6: &str = "ipv6";
 
 /// Cast bytes to `T`.
-unsafe fn cast_ref<T>(bytes: &[u8]) -> &T { unsafe {
-    bytes.as_ptr().cast::<T>().as_ref().unwrap()
-}}
+unsafe fn cast_ref<T>(bytes: &[u8]) -> &T {
+    unsafe { bytes.as_ptr().cast::<T>().as_ref().unwrap() }
+}
 
 /// Cast `T' to bytes.
-unsafe fn cast_bytes<T: Sized>(p: &T) -> &[u8] { unsafe {
-    from_raw_parts(from_ref::<T>(p).cast::<u8>(), size_of::<T>())
-}}
+unsafe fn cast_bytes<T: Sized>(p: &T) -> &[u8] {
+    unsafe { from_raw_parts(from_ref::<T>(p).cast::<u8>(), size_of::<T>()) }
+}
 
 /// Create socket for ioctl communication.
 fn create_socket(address_family: AddressFamily) -> Result<OwnedFd, Errno> {

--- a/src/bsd/mod.rs
+++ b/src/bsd/mod.rs
@@ -67,14 +67,14 @@ static NV_IPV4: &str = "ipv4";
 static NV_IPV6: &str = "ipv6";
 
 /// Cast bytes to `T`.
-unsafe fn cast_ref<T>(bytes: &[u8]) -> &T {
+unsafe fn cast_ref<T>(bytes: &[u8]) -> &T { unsafe {
     bytes.as_ptr().cast::<T>().as_ref().unwrap()
-}
+}}
 
 /// Cast `T' to bytes.
-unsafe fn cast_bytes<T: Sized>(p: &T) -> &[u8] {
+unsafe fn cast_bytes<T: Sized>(p: &T) -> &[u8] { unsafe {
     from_raw_parts(from_ref::<T>(p).cast::<u8>(), size_of::<T>())
-}
+}}
 
 /// Create socket for ioctl communication.
 fn create_socket(address_family: AddressFamily) -> Result<OwnedFd, Errno> {

--- a/src/bsd/nvlist.rs
+++ b/src/bsd/nvlist.rs
@@ -182,7 +182,7 @@ impl<'a> NvList<'a> {
     }
 
     /// Get value for a given `name`.
-    fn get(&self, name: &str) -> Option<&NvValue> {
+    fn get(&self, name: &str) -> Option<&NvValue<'_>> {
         self.items.iter().find(|(n, _)| n == &name).map(|(_, v)| v)
     }
 
@@ -219,7 +219,7 @@ impl<'a> NvList<'a> {
     }
 
     /// Get value as `Vec<NvList>`
-    pub fn get_nvlist_array(&self, name: &str) -> Option<&[NvList]> {
+    pub fn get_nvlist_array(&self, name: &str) -> Option<&[NvList<'_>]> {
         self.get(name).and_then(|value| match value {
             NvValue::NvListArray(array) => Some(array.as_slice()),
             _ => None,
@@ -394,10 +394,14 @@ impl<'a> NvList<'a> {
                 NvValue::Binary(bytes) => buf.extend_from_slice(bytes),
                 NvValue::Bytes(bytes) => buf.extend_from_slice(bytes.as_slice()),
                 NvValue::BoolArray(array) => {
-                    array.iter().for_each(|boolean| buf.push((*boolean).into()));
+                    for boolean in array {
+                        buf.push((*boolean).into());
+                    }
                 }
                 NvValue::NumberArray(array) => {
-                    array.iter().for_each(|number| self.store_u64(*number, buf));
+                    for number in array {
+                        self.store_u64(*number, buf);
+                    }
                 }
                 NvValue::StringArray(array) => {
                     for string in array {

--- a/src/bsd/route.rs
+++ b/src/bsd/route.rs
@@ -1,20 +1,19 @@
 use std::{
     ffi::{CStr, CString},
-    mem::{size_of, MaybeUninit},
+    mem::{MaybeUninit, size_of},
     net::IpAddr,
     os::fd::{AsFd, AsRawFd},
 };
 
 use nix::{
     errno::Errno,
-    sys::socket::{shutdown, socket, AddressFamily, Shutdown, SockFlag, SockType},
+    sys::socket::{AddressFamily, Shutdown, SockFlag, SockType, shutdown, socket},
     unistd::{read, write},
 };
 
 use super::{
-    cast_bytes, cast_ref,
-    sockaddr::{unpack_sockaddr, SockAddrDl, SocketFromRaw},
-    IoError,
+    IoError, cast_bytes, cast_ref,
+    sockaddr::{SockAddrDl, SocketFromRaw, unpack_sockaddr},
 };
 
 // Routing data types are not defined in libc crate, so define then here.

--- a/src/bsd/sockaddr.rs
+++ b/src/bsd/sockaddr.rs
@@ -32,7 +32,7 @@ impl SockAddrIn {
 
 impl SocketFromRaw for SockAddrIn {
     /// Construct `SockAddrIn` from `libc::sockaddr`.
-    unsafe fn from_raw(addr: *const libc::sockaddr) -> Option<Self> {
+    unsafe fn from_raw(addr: *const libc::sockaddr) -> Option<Self> { unsafe {
         if addr.is_null() || (*addr).sa_family != AF_INET {
             None
         } else {
@@ -44,7 +44,7 @@ impl SocketFromRaw for SockAddrIn {
             );
             Some(sockaddr)
         }
-    }
+    }}
 }
 
 impl Default for SockAddrIn {
@@ -126,7 +126,7 @@ impl SockAddrIn6 {
 
 impl SocketFromRaw for SockAddrIn6 {
     /// Construct `SockAddrIn6` from `libc::sockaddr`.
-    unsafe fn from_raw(addr: *const libc::sockaddr) -> Option<Self> {
+    unsafe fn from_raw(addr: *const libc::sockaddr) -> Option<Self> { unsafe {
         if addr.is_null() || (*addr).sa_family != AF_INET6 {
             None
         } else {
@@ -138,7 +138,7 @@ impl SocketFromRaw for SockAddrIn6 {
             );
             Some(sockaddr)
         }
-    }
+    }}
 }
 
 impl Default for SockAddrIn6 {

--- a/src/bsd/sockaddr.rs
+++ b/src/bsd/sockaddr.rs
@@ -5,7 +5,7 @@ use std::{
     ptr::{copy, from_mut},
 };
 
-use super::{cast_bytes, cast_ref, AF_INET, AF_INET6, AF_LINK, SA_IN6_SIZE, SA_IN_SIZE};
+use super::{AF_INET, AF_INET6, AF_LINK, SA_IN_SIZE, SA_IN6_SIZE, cast_bytes, cast_ref};
 
 pub(super) trait SocketFromRaw {
     unsafe fn from_raw(addr: *const libc::sockaddr) -> Option<Self>
@@ -32,19 +32,21 @@ impl SockAddrIn {
 
 impl SocketFromRaw for SockAddrIn {
     /// Construct `SockAddrIn` from `libc::sockaddr`.
-    unsafe fn from_raw(addr: *const libc::sockaddr) -> Option<Self> { unsafe {
-        if addr.is_null() || (*addr).sa_family != AF_INET {
-            None
-        } else {
-            let mut sockaddr: Self = zeroed();
-            copy(
-                addr.cast::<u8>(),
-                from_mut::<Self>(&mut sockaddr).cast::<u8>(),
-                (*addr).sa_len as usize,
-            );
-            Some(sockaddr)
+    unsafe fn from_raw(addr: *const libc::sockaddr) -> Option<Self> {
+        unsafe {
+            if addr.is_null() || (*addr).sa_family != AF_INET {
+                None
+            } else {
+                let mut sockaddr: Self = zeroed();
+                copy(
+                    addr.cast::<u8>(),
+                    from_mut::<Self>(&mut sockaddr).cast::<u8>(),
+                    (*addr).sa_len as usize,
+                );
+                Some(sockaddr)
+            }
         }
-    }}
+    }
 }
 
 impl Default for SockAddrIn {
@@ -126,19 +128,21 @@ impl SockAddrIn6 {
 
 impl SocketFromRaw for SockAddrIn6 {
     /// Construct `SockAddrIn6` from `libc::sockaddr`.
-    unsafe fn from_raw(addr: *const libc::sockaddr) -> Option<Self> { unsafe {
-        if addr.is_null() || (*addr).sa_family != AF_INET6 {
-            None
-        } else {
-            let mut sockaddr: Self = zeroed();
-            copy(
-                addr.cast::<u8>(),
-                from_mut::<Self>(&mut sockaddr).cast::<u8>(),
-                (*addr).sa_len as usize,
-            );
-            Some(sockaddr)
+    unsafe fn from_raw(addr: *const libc::sockaddr) -> Option<Self> {
+        unsafe {
+            if addr.is_null() || (*addr).sa_family != AF_INET6 {
+                None
+            } else {
+                let mut sockaddr: Self = zeroed();
+                copy(
+                    addr.cast::<u8>(),
+                    from_mut::<Self>(&mut sockaddr).cast::<u8>(),
+                    (*addr).sa_len as usize,
+                );
+                Some(sockaddr)
+            }
         }
-    }}
+    }
 }
 
 impl Default for SockAddrIn6 {

--- a/src/bsd/wgio.rs
+++ b/src/bsd/wgio.rs
@@ -1,5 +1,5 @@
 use std::{
-    alloc::{alloc, dealloc, Layout},
+    alloc::{Layout, alloc, dealloc},
     os::fd::AsRawFd,
     ptr::null_mut,
     slice::from_raw_parts,
@@ -8,7 +8,7 @@ use std::{
 use libc::IF_NAMESIZE;
 use nix::{ioctl_readwrite, sys::socket::AddressFamily};
 
-use super::{create_socket, IoError};
+use super::{IoError, create_socket};
 
 // FIXME: `WgReadIo` and `WgWriteIo` have to be declared public.
 ioctl_readwrite!(write_wireguard_data, b'i', 210, WgWriteIo);

--- a/src/key.rs
+++ b/src/key.rs
@@ -6,11 +6,11 @@ use std::{
     str::FromStr,
 };
 
-use base64::{prelude::BASE64_STANDARD, DecodeError, Engine};
+use base64::{DecodeError, Engine, prelude::BASE64_STANDARD};
 #[cfg(feature = "serde")]
 use serde::{
-    de::{Unexpected, Visitor},
     Deserialize, Deserializer, Serialize, Serializer,
+    de::{Unexpected, Visitor},
 };
 use x25519_dalek::{PublicKey, StaticSecret};
 
@@ -221,7 +221,7 @@ mod tests {
     use super::*;
 
     #[cfg(feature = "serde")]
-    use serde_test::{assert_tokens, Token};
+    use serde_test::{Token, assert_tokens};
 
     // Same `Key` in different representations.
     static KEY_B64: &str = "AAECAwQFBgcICQoLDA0OD/Dh0sO0pZaHeGlaSzwtHg8=";

--- a/src/netlink.rs
+++ b/src/netlink.rs
@@ -19,7 +19,6 @@ use netlink_packet_route::{
     rule::{RuleAction, RuleAttribute, RuleFlags, RuleHeader, RuleMessage},
     AddressFamily, RouteNetlinkMessage,
 };
-use netlink_packet_utils::errors::DecodeError;
 use netlink_packet_wireguard::{
     constants::WGPEER_F_REMOVE_ME,
     nlas::{WgDeviceAttrs, WgPeer, WgPeerAttrs},
@@ -50,7 +49,7 @@ pub(crate) enum NetlinkError {
     #[error("Socket error: {0}")]
     SocketError(String),
     #[error("Failed to read response")]
-    ResponseError(#[from] DecodeError),
+    ResponseError(#[from] netlink_packet_core::DecodeError),
     #[error("Netlink payload error: {0}")]
     PayloadError(netlink_packet_core::ErrorMessage),
     #[error("Failed to create WireGuard interface")]
@@ -149,7 +148,7 @@ where
                 message.set_resolved_family_id(*family_id);
             }
             _ => return Err(NetlinkError::UnexpectedPayload),
-        };
+        }
     }
     netlink_request(message, flags, NETLINK_GENERIC)
 }

--- a/src/netlink.rs
+++ b/src/netlink.rs
@@ -2,14 +2,15 @@
 use std::{fmt::Debug, io::ErrorKind, net::IpAddr};
 
 use netlink_packet_core::{
-    NetlinkDeserializable, NetlinkMessage, NetlinkPayload, NetlinkSerializable, NLM_F_ACK,
-    NLM_F_CREATE, NLM_F_DUMP, NLM_F_EXCL, NLM_F_REPLACE, NLM_F_REQUEST,
+    NLM_F_ACK, NLM_F_CREATE, NLM_F_DUMP, NLM_F_EXCL, NLM_F_REPLACE, NLM_F_REQUEST,
+    NetlinkDeserializable, NetlinkMessage, NetlinkPayload, NetlinkSerializable,
 };
 use netlink_packet_generic::{
-    ctrl::{nlas::GenlCtrlAttrs, GenlCtrl, GenlCtrlCmd},
     GenlFamily, GenlMessage,
+    ctrl::{GenlCtrl, GenlCtrlCmd, nlas::GenlCtrlAttrs},
 };
 use netlink_packet_route::{
+    AddressFamily, RouteNetlinkMessage,
     address::{AddressAttribute, AddressMessage},
     link::{InfoKind, LinkAttribute, LinkFlags, LinkHeader, LinkInfo, LinkMessage},
     route::{
@@ -17,23 +18,22 @@ use netlink_packet_route::{
         RouteType,
     },
     rule::{RuleAction, RuleAttribute, RuleFlags, RuleHeader, RuleMessage},
-    AddressFamily, RouteNetlinkMessage,
 };
 use netlink_packet_wireguard::{
+    Wireguard, WireguardCmd,
     constants::WGPEER_F_REMOVE_ME,
     nlas::{WgDeviceAttrs, WgPeer, WgPeerAttrs},
-    Wireguard, WireguardCmd,
 };
 use netlink_sys::{
-    constants::{NETLINK_GENERIC, NETLINK_ROUTE},
     Socket, SocketAddr,
+    constants::{NETLINK_GENERIC, NETLINK_ROUTE},
 };
 use thiserror::Error;
 
 use crate::{
+    IpVersion, Key, WireguardInterfaceError,
     host::{Host, Peer},
     net::IpAddrMask,
-    IpVersion, Key, WireguardInterfaceError,
 };
 
 const SOCKET_BUFFER_LENGTH: usize = 12288;
@@ -207,7 +207,7 @@ where
                         ErrorKind::AlreadyExists => Err(NetlinkError::FileAlreadyExists),
                         ErrorKind::NotFound => Err(NetlinkError::NotFound),
                         _ => Err(NetlinkError::PayloadError(msg)),
-                    }
+                    };
                 }
                 _ => {}
             }

--- a/src/utils.rs
+++ b/src/utils.rs
@@ -10,19 +10,19 @@ use std::{io::Write, process::Stdio};
 #[cfg(not(target_os = "windows"))]
 use std::{net::IpAddr, process::Command};
 
-#[cfg(any(target_os = "freebsd", target_os = "netbsd"))]
-use crate::check_command_output_status;
 #[cfg(not(target_os = "windows"))]
 use crate::Peer;
 use crate::WireguardInterfaceError;
+#[cfg(any(target_os = "freebsd", target_os = "netbsd"))]
+use crate::check_command_output_status;
 #[cfg(any(target_os = "freebsd", target_os = "macos", target_os = "netbsd"))]
 use crate::{
+    IpVersion,
     bsd::{add_gateway, add_linked_route, get_gateway},
     net::IpAddrMask,
-    IpVersion,
 };
 #[cfg(target_os = "linux")]
-use crate::{check_command_output_status, netlink, IpVersion};
+use crate::{IpVersion, check_command_output_status, netlink};
 
 #[cfg(any(target_os = "freebsd", target_os = "linux", target_os = "netbsd"))]
 pub(crate) fn configure_dns(
@@ -321,7 +321,7 @@ pub(crate) fn add_peer_routing(
 ) -> Result<(), WireguardInterfaceError> {
     use nix::errno::Errno;
 
-    use crate::bsd::{delete_gateway, IoError};
+    use crate::bsd::{IoError, delete_gateway};
 
     debug!("Adding peer routing for interface: {ifname}");
     for peer in peers {

--- a/src/wgapi_freebsd.rs
+++ b/src/wgapi_freebsd.rs
@@ -1,11 +1,10 @@
 use std::net::IpAddr;
 
 use crate::{
-    bsd,
+    Host, InterfaceConfiguration, IpAddrMask, Key, Peer, WireguardInterfaceApi,
+    WireguardInterfaceError, bsd,
     utils::{add_peer_routing, clear_dns, configure_dns},
     wgapi::{Kernel, WGApi},
-    Host, InterfaceConfiguration, IpAddrMask, Key, Peer, WireguardInterfaceApi,
-    WireguardInterfaceError,
 };
 
 /// Manages interfaces created with FreeBSD kernel WireGuard module.

--- a/src/wgapi_linux.rs
+++ b/src/wgapi_linux.rs
@@ -1,11 +1,10 @@
 use std::net::IpAddr;
 
 use crate::{
-    netlink,
+    Host, InterfaceConfiguration, IpAddrMask, Key, Peer, WireguardInterfaceApi,
+    WireguardInterfaceError, netlink,
     utils::{add_peer_routing, clean_fwmark_rules, clear_dns, configure_dns},
     wgapi::{Kernel, WGApi},
-    Host, InterfaceConfiguration, IpAddrMask, Key, Peer, WireguardInterfaceApi,
-    WireguardInterfaceError,
 };
 
 /// Manages interfaces created with Linux kernel WireGuard module.

--- a/src/wgapi_userspace.rs
+++ b/src/wgapi_userspace.rs
@@ -13,16 +13,15 @@ use crate::dependencies::check_external_dependencies;
 use crate::netlink;
 #[cfg(any(target_os = "freebsd", target_os = "linux", target_os = "netbsd"))]
 use crate::utils::clear_dns;
-#[cfg(any(target_os = "freebsd", target_os = "macos", target_os = "netbsd"))]
-use crate::{bsd, utils::resolve};
 use crate::{
-    check_command_output_status,
+    Host, InterfaceConfiguration, IpAddrMask, Key, Peer, check_command_output_status,
     error::WireguardInterfaceError,
     utils::{add_peer_routing, configure_dns},
     wgapi::{Userspace, WGApi},
     wireguard_interface::WireguardInterfaceApi,
-    Host, InterfaceConfiguration, IpAddrMask, Key, Peer,
 };
+#[cfg(any(target_os = "freebsd", target_os = "macos", target_os = "netbsd"))]
+use crate::{bsd, utils::resolve};
 
 const USERSPACE_EXECUTABLE: &str = "wireguard-go";
 
@@ -286,7 +285,10 @@ impl WireguardInterfaceApi for WGApi<Userspace> {
                 debug!("Socket removed for interface {}", self.ifname);
             }
             Err(err) if err.kind() == io::ErrorKind::NotFound => {
-                debug!("Socket not found for interface {}, skipping removal as there is nothing to remove. Continuing with further cleanup.", self.ifname);
+                debug!(
+                    "Socket not found for interface {}, skipping removal as there is nothing to remove. Continuing with further cleanup.",
+                    self.ifname
+                );
             }
             Err(err) => {
                 return Err(WireguardInterfaceError::UnixSockerError(format!(
@@ -298,7 +300,10 @@ impl WireguardInterfaceApi for WGApi<Userspace> {
 
         #[cfg(target_os = "macos")]
         {
-            debug!("Clearing DNS entries by applying an empty DNS list to all network services, interface {}", self.ifname);
+            debug!(
+                "Clearing DNS entries by applying an empty DNS list to all network services, interface {}",
+                self.ifname
+            );
             configure_dns(&[], &[])?;
         }
         #[cfg(any(target_os = "linux", target_os = "freebsd", target_os = "netbsd"))]
@@ -366,7 +371,10 @@ impl WireguardInterfaceApi for WGApi<Userspace> {
         );
         match self.read_host() {
             Ok(host) => {
-                debug!("Interface configuration and statistics read successfully for interface {}", self.ifname);
+                debug!(
+                    "Interface configuration and statistics read successfully for interface {}",
+                    self.ifname
+                );
                 trace!("Network information: {host:?}");
                 Ok(host)
             }

--- a/src/wgapi_windows.rs
+++ b/src/wgapi_windows.rs
@@ -10,12 +10,12 @@ use std::{
 };
 
 use crate::{
+    InterfaceConfiguration, WireguardInterfaceApi,
     error::WireguardInterfaceError,
     host::{Host, Peer},
     key::Key,
     net::IpAddrMask,
     wgapi::{Kernel, WGApi},
-    InterfaceConfiguration, WireguardInterfaceApi,
 };
 
 /// Manages interfaces created with Windows kernel using https://git.zx2c4.com/wireguard-nt.
@@ -51,7 +51,9 @@ impl WireguardInterfaceApi for WGApi<Kernel> {
 
         let mut file = File::create(&file_name)?;
 
-        debug!("WireGuard configuration file {file_name} created in {file_path}. Preparing configuration...");
+        debug!(
+            "WireGuard configuration file {file_name} created in {file_path}. Preparing configuration..."
+        );
 
         let address = config
             .addresses
@@ -176,7 +178,9 @@ impl WireguardInterfaceApi for WGApi<Kernel> {
 
                 counter += 1;
             }
-            debug!("Finished waiting for service to be removed, the service is considered to be removed, proceeding further");
+            debug!(
+                "Finished waiting for service to be removed, the service is considered to be removed, proceeding further"
+            );
         }
 
         debug!("Installing the new service for interface {}", self.ifname);
@@ -189,7 +193,9 @@ impl WireguardInterfaceApi for WGApi<Kernel> {
                 WireguardInterfaceError::ServiceInstallationFailed(err.to_string())
             })?;
 
-        debug!("Done installing the new service. Service installation output: {service_installation_output:?}",);
+        debug!(
+            "Done installing the new service. Service installation output: {service_installation_output:?}",
+        );
 
         if !service_installation_output.status.success() {
             let message = format!(
@@ -213,7 +219,9 @@ impl WireguardInterfaceApi for WGApi<Kernel> {
                 WireguardInterfaceError::ServiceInstallationFailed(err.to_string())
             })?;
 
-        debug!("Done disabling automatic restart for the new service. Service update output: {service_update_output:?}",);
+        debug!(
+            "Done disabling automatic restart for the new service. Service update output: {service_update_output:?}",
+        );
         if !service_update_output.status.success() {
             let message = format!(
                 "Failed to configure WireGuard tunnel service: {:?}",

--- a/src/wireguard_interface.rs
+++ b/src/wireguard_interface.rs
@@ -1,6 +1,6 @@
 use std::net::IpAddr;
 
-use crate::{error::WireguardInterfaceError, Host, InterfaceConfiguration, IpAddrMask, Key, Peer};
+use crate::{Host, InterfaceConfiguration, IpAddrMask, Key, Peer, error::WireguardInterfaceError};
 
 /// API for managing a WireGuard interface.
 ///

